src/main/scala/io/iohk/ethereum/consensus/validators/BlockWithCheckpointHeaderValidator.scala

```diff
@@ -157,3 +171,9 @@ class BlockWithCheckpointHeaderValidator(blockchainConfig: BlockchainConfig) {
     else Left(CheckpointHeaderTreasuryOptOutError)
 
 }
+
+object BlockWithCheckpointHeaderValidator {
+  val NoCheckpointInHeaderError: BlockHeaderError = HeaderUnexpectedError(
```

Comment:
Very minor: is it worth having a `HeaderUnexpectedError` only for this? Maybe we could replace this by a `case object NoCheckpointInHeaderError extends BlockHeaderError`?

Reply:
I'm not sure about it. This error is used when `BlockWithCheckpointHeaderValidator` is trying to validate block without a checkpoint, what shouldn't have happened and is something unexpected. Maybe it's my subjective feeling, but when I did this change, I started to think that I promote that unexpected error to something which is expected negative behaviour. WDYT?

Reply:
Hmm I think we can leave it as is then